lib/linter/linter.js

```diff
@@ -314,6 +317,62 @@ function createLintingProblem(options) {
     };
 }
 
+/**
+ * Wraps the value in an Array if it isn't already one.
+ * @template T
+ * @param {T|T[]} value Value to be wrapped.
+ * @returns {Array} The value as an array.
+ */
+function asArray(value) {
+    return Array.isArray(value) ? value : [value];
+}
+
+/**
+ * Pushes a problem to inlineConfigProblems if ruleOptions are redundant.
+ * @param {ConfigData} config Provided config.
+ * @param {Object} loc A line/column location
+ * @param {Array} problems Problems that may be added to.
+ * @param {string} ruleId The rule ID.
+ * @param {Array} ruleOptions The rule options, merged with the config's.
+ * @param {Array} ruleOptionsInline The rule options from the comment.
+ * @param {"error"|"warn"} severity The severity to report.
+ * @returns {void}
+ */
+function addProblemIfSameSeverityAndOptions(config, loc, problems, ruleId, ruleOptions, ruleOptionsInline, severity) {
+    const existingConfigRaw = config.rules?.[ruleId];
+    const existingConfig = existingConfigRaw ? asArray(existingConfigRaw) : ["off"];
+    const existingSeverity = normalizeSeverityToString(existingConfig[0]);
+    const inlineSeverity = normalizeSeverityToString(ruleOptions[0]);
+    const sameSeverity = existingSeverity === inlineSeverity;
+
+    if (!sameSeverity) {
+        return;
+    }
+
+    const alreadyConfigured = existingConfigRaw
+        ? `is already configured to '${existingSeverity}'`
+        : "is not enabled so can't be turned off";
+    let message;
+
+    if (existingConfig.length === 1 && ruleOptions.length === 1) {
```

Comment:
```suggestion
    if ((existingConfig.length === 1 && ruleOptions.length === 1) || existingSeverity === "off") {
```

I think rule severity `"off"` should be a special case where options don't matter.